src/coreclr/jit/morph.cpp

```diff
@@ -11096,25 +11100,33 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
             }
         }
 
-        FieldSeqNode* srcFldSeq   = nullptr;
-        unsigned      srcLclNum   = BAD_VAR_NUM;
-        LclVarDsc*    srcLclVar   = nullptr;
-        bool          srcDoFldAsg = false;
+        FieldSeqNode* srcFldSeq    = nullptr;
+        unsigned      srcLclNum    = BAD_VAR_NUM;
+        LclVarDsc*    srcLclVar    = nullptr;
+        unsigned      srcLclOffset = 0;
+        bool          srcDoFldAsg  = false;
+
+        bool srcUseLclFld  = false;
+        bool destUseLclFld = false;
 
         if (src->IsLocal())
         {
             srcLclVarTree = src->AsLclVarCommon();
             srcLclNum     = srcLclVarTree->GetLclNum();
             if (src->OperGet() == GT_LCL_FLD)
             {
-                srcFldSeq = src->AsLclFld()->GetFieldSeq();
+                srcFldSeq    = src->AsLclFld()->GetFieldSeq();
+                srcLclOffset = src->AsLclFld()->GetLclOffs();
             }
         }
         else if (src->OperIsIndir())
         {
             if (src->AsOp()->gtOp1->IsLocalAddrExpr(this, &srcLclVarTree, &srcFldSeq))
             {
                 srcLclNum = srcLclVarTree->GetLclNum();
+
+                srcLclVar    = &lvaTable[srcLclNum];
```

Comment:
Looks like setting `srcLclVar` here is unnecessary (it will get set below). Or, perhaps you want to use it just below as `srcLclOffset = srcLclVar->GetLclOffs();`? But shouldn't the values be the same?